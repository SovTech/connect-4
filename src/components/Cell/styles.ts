import styled from 'styled-components';
import theme from '../../theme';

const cellSize = '30px';

export const StyledCell = styled.button`
  background: ${theme.disabledColorLight};
  border-radius: 50%;
  border: none;
  color: #333;
  cursor: pointer;
  display: inline-block;
  font-size: 12px;
  height: ${cellSize};
  margin: 10px auto;
  outline: none;
  padding: 0;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  width: ${cellSize};

  &:hover {
    background-color: ${theme.disabledColor};
  }

  &.red {
    background-color: ${theme.error} !important;
  }

  &.yellow {
    background-color: ${theme.warning} !important;
  }
`;
