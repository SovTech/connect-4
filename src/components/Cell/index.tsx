import classnames from 'classnames';
import * as React from 'react';
import { StyledCell } from './styles';

type Props = {
  x: number;
  y: number;
  cell: string;
  nextPlayer: string;
  addPiece: Function;
}

export default class Cell extends React.Component<Props> {
  handleAddPiece = () => {
    this.props.addPiece(this.props.y, this.props.nextPlayer);
  };

  render() {
    const cellClasses = classnames({
      'red': (this.props.cell === 'red'),
      'yellow': (this.props.cell === 'yellow')
    });

    return (
      <StyledCell
        className={cellClasses}
        onClick={this.handleAddPiece}
      />
    );
  }

}
