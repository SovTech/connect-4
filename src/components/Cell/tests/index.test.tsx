import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Cell from '../index';

it('renders a Cell correctly', () => {
  const tree = renderer
    .create(<Cell cell='' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
