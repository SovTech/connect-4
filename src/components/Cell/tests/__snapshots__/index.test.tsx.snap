// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Cell correctly 1`] = `
<button
  className="sc-bdVaJa iDzsfT"
/>
`;
